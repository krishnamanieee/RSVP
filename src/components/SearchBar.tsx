import React from 'react';
import {View} from 'react-native';
import TextInput from './TextInput';

export interface ISearchBarProps {
  value: string;
  onValueChange: (value: string) => void;
}

class SearchBar extends React.PureComponent<ISearchBarProps, any> {
  render(): React.ReactElement {
    const {value, onValueChange} = this.props;
    return (
      <View>
        <TextInput
          placeholder={'Search the participant'}
          value={value}
          onChangeText={onValueChange}
        />
      </View>
    );
  }
}
export default SearchBar;
