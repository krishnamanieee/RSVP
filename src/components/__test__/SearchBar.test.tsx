import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar, {ISearchBarProps} from '../SearchBar';
let props: ISearchBarProps = {
  onValueChange: jest.fn(),
  value: '',
};
describe('SearchBar', () => {
  it('should match match snapshot', () => {
    const tree = renderer.create(<SearchBar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
