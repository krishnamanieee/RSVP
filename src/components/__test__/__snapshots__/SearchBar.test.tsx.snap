// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should match match snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "marginHorizontal": 5,
        "marginVertical": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[MockFunction]}
      placeholder="Search the participant"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#dadada",
            "paddingHorizontal": 5,
            "paddingVertical": 10,
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
